examples/bovw/plot_bovw.py

```diff
@@ -0,0 +1,141 @@
+"""
+===================
+Bag of Visual Words
+===================
+
+An illustration of a Bag of Visual Words approach for image recognition
```

Comment:
Maybe explain that this is not how one would actually solve this problem but that it's simplified not use any vision library and to run fast enough ;)